File: API/App.Model/Entities/MatchPoint.cs
using App.Model.Entities.Common;
using App.Model.Enums;
using System.ComponentModel.DataAnnotations;

namespace App.Model.Entities
{
    public class MatchPoint : AuditableEntity
    {
        public Match Match { get; set; }
        public Player GoalScorer { get; set; }
        public Team Team { get; set; }

        [Required]
        public MatchPointType Point { get; set; }

        [Required]
        public int MinuteOfMatch { get; set; }
    }
}

File: WebUI/app-frontend/src/app/models/match/match-point.ts
import { MatchPointType } from "src/app/enums/match-point-type";

export class MatchPoint {
    public id : string;
    public isActive : boolean;
    public matchId : string;
    public playerId : string;
    public point : MatchPointType;
    public minuteOfMatch : number;
}
